core/templates/dev/head/filters.js

```diff
@@ -240,6 +240,14 @@ oppia.filter('parameterizeRuleDescription', [
           replacementText = inputs[varName];
         }
 
+        if (typeof (replacementText) === 'string') {
+          // Replacing $ with $$. When this is used in the next replace
+          // statement, it will be converted back to $.
+          // Also to replace by $$, the appropriate string is $$$$
+          // as $$ -> $
+          replacementText = replacementText.replace('$', '$$$$');
```

Comment:
Also, was there any specific reason to use 4 (not 3)?

Reply:
Thanks for pointing that out. I changed it to 3($) as that would also work. And also the issue was replace was substituting only the first location. changed it to match /\$/g. Now it works for multiple $ symbols also. Changed the karma test to take care of this too.